### PR TITLE
 Implement IsModifiableModule and update J9JvmtiErrorMapping 

### PR DESCRIPTION
**Pull request by babsingh**
New JVMTI error code, JVMTI_ERROR_UNMODIFIABLE_MODULE, has been defined.

jvmtiGetErrorName doesn't return name of new JVMTI error codes. Updated J9JvmtiErrorMapping to include new JVMTI error codes: JVMTI_ERROR_INVALID_MODULE and JVMTI_ERROR_UNMODIFIABLE_MODULE.

Implemented JVMTI function IsModifiableModule. This function determines whether a module is modifiable.

Signed-off-by: Babneet Singh <sbabneet@ca.ibm.com>

### CODE REVIEW

**Comment by pshipton** [comment]
jenkins test sanity

**Comment by pshipton** [comment]
@tajila any concerns?

File: runtime/jvmti/jvmtiModules.c [hunk 1]
```diff
@@ -686,3 +686,70 @@ jvmtiAddModuleProvides(jvmtiEnv* jvmtiEnv,
...
+			moduleJ9Class = J9VMJAVALANGREFLECTMODULE_OR_NULL(vm);
+		} else {
+			moduleJ9Class = J9VMJAVALANGMODULE_OR_NULL(vm);
+		}
+
+		Assert_JVMTI_notNull(moduleJ9Class);
+
+		if (!isSameOrSuperClassOf(moduleJ9Class, J9OBJECT_CLAZZ(currentThread, moduleObject))) {
+			rc = JVMTI_ERROR_INVALID_MODULE;
+		} else if (J9_IS_J9MODULE_UNNAMED(vm, J9OBJECT_ADDRESS_LOAD(currentThread, moduleObject, vm->modulePointerOffset))) {
```
Comment (tajila) [inline runtime/jvmti/jvmtiModules.c]: maybe comment out this check since and only have the else case since we will always be returning true

Comment (pshipton) [reply runtime/jvmti/jvmtiModules.c]: I'll just merge it the way it is. This condition is explicitly called out in the java.lang.instrument.Instrumentation.isModifiableModule​() javadoc, so the condition needs to be in the code if there is ever a case that returns false. Keeping the condition avoids having to slow down the code later.

**Pull request merged**